build/update-locales.sh

```diff
@@ -1,14 +1,49 @@
-#!/bin/sh
+#!/bin/bash
+
+set -e
+
+SED=sed
+
+if [[ $OSTYPE == 'darwin'* ]]; then
+  # for macOS developers, use "brew install gnu-sed"
+  SED=gsed
+fi
```

Comment:
Could just do the same detection that `Makefile` does and adapt syntax based on whether it is GNU or BSD sed. Better than to require the installation of additional software :)

Reply:
I'll check it out.

Reply:
https://github.com/go-gitea/gitea/pull/23254

Reply:
Thank you. Actually I have thought about this problem, but I didn't choose the `sed -i''` for BSD sed, because the script has been here for long time, nobody really needs it, I just want to avoid any inconsistency. And I think this PR is the last fix by modifying the script 😂 in the future  we should decide whether use Go to rewrite or totally replace the ini package.

The `sed -i''` might also work, just like it in Makefile, but  I haven't tested it.

Reply:
Some more explanations about my "worry about inconsistency": this script is hard to debug, and not covered by any  test (unlike the sed in Makefile, any error can be found in the first time and gets fixed quickly). If there is any inconsistency between GNU sed and BSD sed, the problem can not be found until next crowdin sync runs, and it's difficult for macOS users to debug the inconsistency if  they are using a different version.

If there is no inconsistency, then using `sed -i''` is fine for BSD sed.

Just share some of my thoughts, for your information, no objection nor block. 😁